diag_manager/fms_diag_object.F90

```diff
@@ -73,14 +89,20 @@ subroutine fms_diag_object_init (obj,diag_subset_output)
 #ifdef use_yaml
  if (obj%initialized) return
 
-!TODO: allocate the file, field, and buffer containers
 ! allocate(diag_objs(get_num_unique_fields()))
   CALL diag_yaml_object_init(diag_subset_output)
   CALL fms_diag_axis_object_init()
   obj%files_initialized = fms_diag_files_object_init(obj%FMS_diag_files)
   obj%fields_initialized = fms_diag_fields_object_init (obj%FMS_diag_fields)
-  obj%registered_variables = 0
-  obj%initialized = .true.
+  obj%buffers_initialized = fms_diag_buffer_init(obj%FMS_diag_buffers_0d, 0) .and. &
+                            fms_diag_buffer_init(obj%FMS_diag_buffers_1d, 1) .and. &
+                            fms_diag_buffer_init(obj%FMS_diag_buffers_2d, 2) .and. &
+                            fms_diag_buffer_init(obj%FMS_diag_buffers_3d, 3) .and. &
+                            fms_diag_buffer_init(obj%FMS_diag_buffers_4d, 4) .and. &
+                            fms_diag_buffer_init(obj%FMS_diag_buffers_5d, 5)
+                             
+ registered_variables = 0
```

Comment:
You need to pull from dmUpdate, this should be obj%registered_variables = 0
and the `registered_variables` should be removed